packages/editor/src/components/block-compare/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { castArray } from 'lodash';
+import { diffChars } from 'diff';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { getBlockType, getSaveContent, getSaveElement } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import BlockView from './block-view';
+
+class BlockCompare extends Component {
+	getDifference( originalContent, newContent ) {
+		const difference = diffChars( originalContent, newContent );
+
+		return difference.map( ( item, pos ) => {
+			const classes = classnames( {
+				'editor-block-compare__added': item.added,
+				'editor-block-compare__removed': item.removed,
+			} );
+
+			return <span key={ pos } className={ classes }>{ item.value }</span>;
+		} );
+	}
+
+	getOriginalContent( block ) {
+		// Get current block details
+		const blockType = getBlockType( block.name );
+
+		return {
+			rawContent: block.originalContent,
+			renderedContent: getSaveElement( blockType, block.attributes ),
```

Comment:
I've just given this another test, and I'm not sure the original content's preview displays quite what I was expecting. My steps:

1. Add some simple text to a paragraph block
2. Edit the html and add an inline style `style="font-size:25px;"`
3. View the comparison
4. Note that the original content preview doesn't show the font size at 25px, which is what I'd get if I selected 'Keep as HTML'.

To make the preview display correctly I tried using RawHTML like the HTML block does for its preview, which seemed to do the trick:
```
	getOriginalContent( block ) {
		return {
			rawContent: block.originalContent,
			renderedContent: (
				<RawHTML>{ block.originalContent }</RawHTML>
			),
		};
	}
```

Reply:
Interesting!

RawHTML uses `dangerouslySetInnerHTML`, which opens up things like `<p onClick="alert(1)">fdsfsdfds fd sfds</p>`. I'm not sure if this is a problem in this context and what the behaviour is supposed to be so I'll have a poke around.

Reply:
The 'preview' mode of the HTML block uses the `<Sandbox />` component to display arbitrary HTML, which neutralises my above example, and seems to be the way to go. The styling is different so I'll see if it can be made to fit in.

Reply:
Hmm, yep, thanks for looking into it. Lets go with what we've got and then maybe look into this on a separate issue.